oap-server/server-bootstrap/src/main/resources/lal/envoy-als.yaml

```diff
@@ -0,0 +1,39 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+rules:
+  - name: envoy-als
+    dsl: |
+      filter {
+        json {
+        }
+        // only collect error logs
+        if (parsed?.response?.responseCode < 300 && !parsed?.commonProperties?.responseFlags) {
```

Comment:
I think this still is not perfect. If there is many upstream error mixing with some internal error, most likely those logs wouldn't be collected ever.
I think we need 2 `rateLimit` sampling branches?

Reply:
Now I use `service:internalErrorCode` and `service:httpResponseCode` as sampler key, that is, we will sample 100 qps per service per internalErrorCode if internalErrorCode is not empty, or (if internalErrorCode is empty), sample 100 qps per service per httpResponseCode if httpResponseCode >= 300